src/handler/mod.rs

```diff
@@ -0,0 +1,143 @@
+/// This module contains most of the low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater than 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) => {
+                if t <= 0 {
+                    bail!("countdown ended, check greenboot-rollback status")
+                }
+            }
+            None => bail!("boot_counter is not set"),
+        }
+    }
+    log::info!("restarting system");
+    let status = Command::new("systemctl").arg("reboot").status()?;
+    if status.success() {
+        return Ok(());
+    }
+    bail!("systemd returned error");
+}
+
+/// rollback to previous ostree deployment if boot counter is less than 0
+pub fn handle_rollback() -> Result<()> {
+    if let Some(t) = get_boot_counter() {
+        if t <= 0 {
+            log::info!("Greenboot will now attempt to rollback");
+            let status = Command::new("rpm-ostree").arg("rollback").status()?;
+            if status.success() {
+                return Ok(());
+            }
+            bail!(status.to_string());
+        }
+    }
+    log::info!("Rollback not initiated as boot_counter is either unset or not equal to 0");
+    Ok(())
+}
+
+/// sets grub variable boot_counter if not set
+pub fn set_boot_counter(reboot_count: i32) -> Result<()> {
+    if let Some(current_counter) = get_boot_counter() {
+        log::info!("boot_counter={current_counter}");
+        return Ok(());
+    } else if set_grub_var("boot_counter", reboot_count) {
```

Comment:
you don't need the `else if` as you're returning from the previous `if let` branch